File: silbot3_tutorials/src/device/wheelencoder_subscribe.cpp
#include "ros/ros.h"
#include "silbot3_msgs/Device_WheelEncoder_Msg.h"

void callback(const silbot3_msgs::Device_WheelEncoder_Msg::ConstPtr& msg)
{
  ROS_INFO("x : %d, y : %d, theta : %d", msg->x, msg->y,msg->t);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "wheelencoder_listener_cpp");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("/DeviceNode/WheelEncoder/data", 1000, callback);

  ros::spin();

  return 0;
}

File: scripts/device/wheelencoder_subscribe.py
#! /usr/bin/env python
import rospy
from silbot3_msgs.msg import Device_WheelEncoder_Msg

def callback(msg) :
    x = msg.x
    y = msg.y
    theta = msg.t
    rospy.loginfo("x : " + str(x) +", y : "+str(y) + " theta : " + str(theta))

def listen_touch() :
    subscriber = rospy.Subscriber('/DeviceNode/WheelEncoder/data', Device_WheelEncoder_Msg, callback)

if __name__ == "__main__" :
    rospy.init_node("wheelencoder_listener", anonymous=True)
    listen_touch()
    rospy.spin()
